src/java.base/share/classes/sun/net/www/http/HttpClient.java

```diff
@@ -428,8 +428,7 @@ protected boolean available() {
             try {
                 old = serverSocket.getSoTimeout();
                 serverSocket.setSoTimeout(1);
-                BufferedInputStream tmpbuf =
-                        new BufferedInputStream(serverSocket.getInputStream());
+                InputStream tmpbuf = serverSocket.getInputStream();
                 int r = tmpbuf.read();
                 if (r == -1) {
                     logFinest("HttpClient.available(): " +
```

Comment:
There are some subtle things going on there. Using a BufferedInputStream ensure that all the bytes available on the socket will be read, up to the buffer capacity. Can you revert this change? I'd rather that this clean up be handled separately. I have logged https://bugs.openjdk.java.net/browse/JDK-8263599.

Reply:
Sure, I'll revert this.